### PR TITLE
DBZ-8307 Fix loss data issue for RabbitMQ

### PR DESCRIPTION
**Pull request by Qwedir**
Done based on the principle of: https://github.com/debezium/debezium-server/pull/96

Link issue:
https://issues.redhat.com/browse/DBZ-8307

### CODE REVIEW

**Comment by Qwedir** [comment]
@jpechane I understand and see the issue. 
Here’s what I suggest as a solution. The code compiles, but I haven’t tested this solution.

File: debezium-server-rabbitmq/src/main/java/io/debezium/server/rabbitmq/RabbitMqStreamNativeChangeConsumer.java [hunk 1]
```diff
@@ -230,17 +235,41 @@ public void handleBatch(List<ChangeEvent<Object, Object>> records, RecordCommitt
...
         }
 
-        LOGGER.trace("Sent messages");
-        committer.markBatchFinished();
+        if (!hasError.get()) {
+            LOGGER.trace("All messages sent successfully");
+            committer.markBatchFinished();
+        }
+        else {
+            LOGGER.error("Batch processing was incomplete due to record processing errors.");
```
Comment (jpechane) [inline debezium-server-rabbitmq/src/main/java/io/debezium/server/rabbitmq/RabbitMqStreamNativeChangeConsumer.java:271]: I believe an exception should be thrown here.

File: debezium-server-rabbitmq/src/main/java/io/debezium/server/rabbitmq/RabbitMqStreamNativeChangeConsumer.java [hunk 2]
```diff
@@ -230,17 +235,41 @@ public void handleBatch(List<ChangeEvent<Object, Object>> records, RecordCommitt
...
+                            }
                         });
             }
             catch (StreamException e) {
                 throw new DebeziumException(e);
             }
+        }
 
-            committer.markProcessed(record);
+        if (!latch.await(producerConfirmTimeout, TimeUnit.SECONDS)) {
```
Comment (jpechane) [inline debezium-server-rabbitmq/src/main/java/io/debezium/server/rabbitmq/RabbitMqStreamNativeChangeConsumer.java:261]: IMHO we need a new config option for this wait set to a larger default value as timeout for sending a single message could be different to sending a batch.

**Review by jpechane — COMMENTED** [review]
@Qwedir Nice work, I left two addiitional comments but I beleive the implementation as scuh is correct.

File: debezium-server-rabbitmq/src/main/java/io/debezium/server/rabbitmq/RabbitMqStreamNativeChangeConsumer.java (hunk 2 above)
Comment (Qwedir) [reply debezium-server-rabbitmq/src/main/java/io/debezium/server/rabbitmq/RabbitMqStreamNativeChangeConsumer.java]: done

File: debezium-server-rabbitmq/src/main/java/io/debezium/server/rabbitmq/RabbitMqStreamNativeChangeConsumer.java (hunk 1 above)
Comment (Qwedir) [reply debezium-server-rabbitmq/src/main/java/io/debezium/server/rabbitmq/RabbitMqStreamNativeChangeConsumer.java]: done

**Pull request merged**